manifests/pub_dir.pp

```diff
@@ -16,13 +16,19 @@
 
   ensure_packages('katello-client-bootstrap')
 
+  if ($foreman_proxy_content::pulp_master or $foreman_proxy_content::pulp) {
+    $additional_includes = ["${apache::confd_dir}/pulp-vhosts80/*.conf"]
+  } else {
+    $additional_includes = []
+  }
+
   apache::vhost { 'foreman_proxy_content':
```

Comment:
I often wondered about this and it doesn't really make sense to me.

On a regular Katello install, there's Foreman listening on port 80 with the same servername (as @evgeni kindly provided):
```
[root@centos7-katello-3-16 httpd]# httpd -S
VirtualHost configuration:
*:5000                 centos7-katello-3-16.yatsu.example.com (/etc/httpd/conf.d/03-crane.conf:6)
*:443                  centos7-katello-3-16.yatsu.example.com (/etc/httpd/conf.d/05-foreman-ssl.conf:6)
*:80                   is a NameVirtualHost
         default server centos7-katello-3-16.yatsu.example.com (/etc/httpd/conf.d/05-foreman.conf:6)
         port 80 namevhost centos7-katello-3-16.yatsu.example.com (/etc/httpd/conf.d/05-foreman.conf:6)
                 alias foreman
         port 80 namevhost centos7-katello-3-16.yatsu.example.com (/etc/httpd/conf.d/05-foreman_proxy_content.conf:6)
```

There `/pub` is provided by https://github.com/theforeman/puppet-katello/blob/7200bf3152230ac938fd9b088c005c1f13ed38c9/templates/pulp-apache.conf.erb#L5-L11

So on a Katello install, this vhost is redundant and only confusing.

I don't have the data for a content proxy, but there we have Pulp 2 running instead of a Foreman and this code is triggered:
https://github.com/theforeman/puppet-foreman_proxy_content/blob/bd6654b038c6dc41518fa59469808a4a28a88a73/templates/httpd_pub.erb#L1-L9

So effectively, this vhost is only relevant on a machine where there is no Foreman and no Pulp 2.

In https://github.com/theforeman/puppet-foreman_proxy_content/ I put `/pub` in the Pulp 3 vhost.

So effectively I think what you can do is leave out `additional_includes` altogether. Then you can move `include foreman_proxy_content::pub_dir` to an else branch in `init.pp`.

The only thing I wonder about is the `ensure_packages('katello-client-bootstrap')` but I wonder if that's really needed.

Reply:
In https://github.com/theforeman/puppet-foreman_proxy_content/pull/283 I implemented that.